app/controllers/users/two_factor_setup_controller.rb

```diff
@@ -0,0 +1,55 @@
+module Users
+  class TwoFactorSetupController < TwoFactorVerificationController
+    before_action :authenticate_user!
+    before_action :confirm_two_factor!
+    before_action :verify_direct_otp_sent, only: [:confirm, :update]
+
+    def new
+    end
+
+
+    def create
+      current_user.unconfirmed_phone_number =  params[:phone_number]
+
+      if current_user.save
+        current_user.send_new_direct_otp_for(:unconfirmed_phone_number)
+        redirect_to confirm_users_two_factor_setup_path and return
+      end
+
+      render :new
```

Comment:
Stylistically, we've tended to put the `render :new` in an else clause rather than having an early return.